index.html

```
@@ -453,17 +427,12 @@ <h2>6. Send the Message:</h2>
                             output('salt',
                                 base64url.encode(crypto.getRandomValues(
                                                         new Uint8Array(16))));
-                            try {
-                                // Auth key allows for improved Crypto-Key support.
-                                // However, it's being slowly rolled out.
-                                let authKey = subscription_info.getKey('auth');
-                                output('authKey', base64url.encode(authKey));
-                                if (! authKey) {
-                                    console.warn ("This browser doesn't support auth key. Please update.");
-                                }
-                            } catch (e) {
-                                //throw err;
-                                console.warn("Crypto-Key support unavailable.");
+                            // Auth key allows for improved Crypto-Key support.
+                            // However, it's being slowly rolled out.
+                            let authKey = subscription_info.getKey('auth');
```

Heh, I really ought to just drop this.
Chrome will never include the same Crypto-Key support, and it's one of the main reasons that this page only works with Gecko browsers.

---

> Chrome will never include the same Crypto-Key support, and it's one of the main reasons that this page only works with Gecko browsers.

I'm not sure, what does Chrome do ? Because the auth is required now

---

It's more what it doesn't do.
It's been a while since I've worked on this code (clearly), but IIRC, `webCrypto` on Chrome doesn't support `importKey` in `raw` form. The original idea was that folk could generate their own keys externally, paste them into the page, and then see the results. Folk have never done that (to my knowledge). 

I should make the field read-only, and update the `preflight()` function to:
1) use `async` instead of promise calls
2) not try the `export` / `import` cycle and replace it with simple checks for `ECDH`. 

As it stands, the `preflight` check is failing on Chrome because `webCrypto.generateKey()` silently fails without throwing an exception, but it's probably because the API changed over the past decade.